lib/rucio/web/ui/common/utils.py

```diff
@@ -20,6 +20,51 @@
 from rucio.common.config import config_get
 from rucio.db.sqla.constants import AccountType
 
+try:
+    AUTH_TYPE = config_get('webui', 'auth_type')
+    if AUTH_TYPE == 'saml':
+        from onelogin.saml2.auth import OneLogin_Saml2_Auth
+except:
+    AUTH_TYPE = 'x509'
+
+
+def prepare_webpy_request(request, data):
+    """
+    Prepare a webpy request for SAML
+
+    :param request: webpy request object
+    :param data: GET or POST data
+    """
+    return {
+        'https': 'on' if request['wsgi.url_scheme'] == 'https' else 'off',
```

Comment:
i'm strongly in favour of failing hard if https is not available.

Reply:
So if https is not available, an error should be raised?

Reply:
ah wait, this is for everything, or only for the authentication? authentication must be encrypted, the rest doesn't matter

Reply:
This function is used to create the request format for SAML login. So this is used to login to IdP